bin/citrea/tests/evm/ethers.js_tests/test.js

```diff
@@ -0,0 +1,340 @@
+        expect(count > 0).to.be.true;
+    });
+
+    it("getTransactionReceipt returns the correct receipt", async function() {
+        let receipt = await provider.getTransactionReceipt(first_tx_receipt.hash);
+        expect(receipt.hash).to.equal(first_tx_receipt.hash);
+        expect(receipt.blockNumber).to.equal(first_tx_receipt.blockNumber);
+        expect(receipt.index).to.equal(first_tx_receipt.index);
+    });
+
+    it("call on non-existent function errors with the correct message", async function() {
+        const abi = [
+            {
+                "type": "function",
+                "name": "ERRENOUS_FUNC",
+                "inputs": [],
+                "outputs": [
+                  {
+                    "name": "",
+                    "type": "address",
+                    "internalType": "address"
+                  }
+                ],
+                "stateMutability": "view"
+              },
+        ];
+
+        const contractAddress = '0x3100000000000000000000000000000000000001';
+        const contract = new ethers.Contract(contractAddress, abi, provider);
+
+        let tx = {
+            to: "0x3100000000000000000000000000000000000001",
+            data: contract.interface.encodeFunctionData('ERRENOUS_FUNC', []),
+            chainId: 5655,
+        };
+
+        try {
+            await provider.call(tx);
+            expect.fail('Expected an error to be thrown');
+        } catch (error) {
+            expect(error.message).to.equal('missing revert data (action="call", data=null, reason=null, transaction={ "data": "0xd6c7a27a", "to": "0x3100000000000000000000000000000000000001" }, invocation=null, revert=null, code=CALL_EXCEPTION, version=6.12.1)');
+        }        
+    });
+
+    it("call with wrong function parameters errors with the correct message", async function() {
+        this.timeout(0);
+        
+        const abi = [
+            {
+                "type": "function",
+                "name": "withdraw",
+                "inputs": [
+                    {
+                        "name": "bitcoin_address",
+                        "type": "bytes32",
+                        "internalType": "bytes32"
+                    }
+                ],
+                "outputs": [],
+                "stateMutability": "payable"
+            },
+        ];
+    
+        const contractAddress = '0x3100000000000000000000000000000000000002';
+        let wallet = new ethers.Wallet('0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80', provider);
+        const contract = new ethers.Contract(contractAddress, abi, wallet);
+        const bitcoinAddress = ethers.encodeBytes32String('bc1qa0a0a0a0a0a0a0a0a0a0a0a0');
+    
+        let tx = {
+            to: contractAddress,
+            value: ethers.parseEther('0.9'),
+            data: contract.interface.encodeFunctionData('withdraw', [bitcoinAddress]),
+            from: wallet.address
+        };
+
+        try {
```

Comment:
isnt there a better parsed ethers.js error object?

does it really just throw the erros like this?

have you tried the same thing on another rpc or reth?